lib/src/domain/usecase/fetch_genres_usecase.dart

```diff
@@ -1,21 +1,38 @@
-import 'package:http/http.dart';
+import 'package:connectivity_plus/connectivity_plus.dart';
 
 import '../../core/usecase/i_usecase.dart';
 import '../../core/util/data_state.dart';
 import '../../core/util/enums.dart';
-import '../../data/repository/api_repository.dart';
+import '../../data/local/movie_database.dart';
 import '../entity/genre.dart';
+import '../entity/movie.dart';
 import '../repository/i_my_repository.dart';
 
 class FetchGenresUseCase
-    extends IUseCase<Future<DataState<List<Genre>>>, List<int>> {
+    extends IUseCase<Future<DataState<List<Genre>>>, Movie> {
   final IMyRepository repository;
+  final MovieDatabase database;
 
-  FetchGenresUseCase({IMyRepository? repository})
-      : repository = repository ?? APIRepository(client: Client());
+  FetchGenresUseCase({
+    required this.repository,
+    required this.database,
```

Comment:
The database instance must be in the repository, let's make a rafactor

Reply:
Big refactor done!